File: src/main/java/maxdistructo/droidbot2/commands/Debug.java
package maxdistructo.droidbot2.commands;

import de.btobastian.sdcf4j.Command;
import de.btobastian.sdcf4j.CommandExecutor;

import java.util.List;

import maxdistructo.droidbot2.background.Perms;
import sx.blah.discord.handle.obj.*;

public class Debug implements CommandExecutor {
    //@Command(aliases = {"/debug"}, description = "Shows debug info for making code for the bot.", usage = "/debug")
    public static String onDebugCommand(String[] args, IMessage message) {
        IUser author = message.getAuthor();
        IChannel channel = message.getChannel();
        IGuild guild = message.getGuild();
        IUser owner = guild.getOwner();
        List<IRole> roles = guild.getRoles();


        if(Perms.checkMod(message)) {
            return "You are: " + author + ". \n" + "Your channel is: " + channel + ". \n" + "Your Owner is: " + owner + ". \n" + "Your server's roles are: " + roles;
        }
        else{
            return "You have insufficient perms to use this command";
        }

    }
}

File: src/main/java/maxdistructo/discord/bots/droidbot/commands/Debug.kt
package maxdistructo.discord.bots.droidbot.commands


import maxdistructo.discord.bots.droidbot.background.PrivUtils
import maxdistructo.discord.core.Perms
import maxdistructo.discord.core.command.BaseCommand
import sx.blah.discord.handle.obj.IMessage

class Debug : BaseCommand() {
    override val commandName: String
        get() = "debug"
    override val helpMessage: String
        get() = "debug - Debug information"
    override val requiresAdmin: Boolean
        get() = true

    override fun init(message: IMessage, args: List<String>): String {
       return onDebugCommand(PrivUtils.listToArray(args), message)
    }

    fun onDebugCommand(args: Array<String>, message: IMessage): String {
        val author = message.author
        val channel = message.channel
        val guild = message.guild
        val owner = guild.owner
        val roles = guild.roles


        return if (Perms.checkMod(message)) {
            "You are: $author. \nYour channel is: $channel. \nYour Owner is: $owner. \nYour server's roles are: $roles"
        } else {
            "You have insufficient perms to use this command"
        }

    }
}
